src/main/java/wtf/g4s8/examples/system/ResourceManager.java

```diff
@@ -0,0 +1,20 @@
+package wtf.g4s8.examples.system;
+
+import wtf.g4s8.examples.spaxos.Acceptor;
+
+import java.util.List;
+
+public interface ResourceManager {
+    //check if update is doable.
+    //Yes -> prepare
+    //No -> abort
+    void update(Patch patch);
+
+    void commit(Patch patch);
```

Comment:
@stepanvalyavskiy commit doesn't accept patches to apply, it accepts only transaction id